File: app/src/at-tamle/java/asiantech/internship/winter/ex21/totalanalytical.java
package asiantech.internship.winter.ex21;

import java.util.Scanner;
//  tinh tong + phan tich thua so ngto
public class totalanalytical {
    public static int process(){
        Scanner input= new Scanner(System.in); boolean check= false;
        int n=0; while(!check){
            System.out.print(" "); try{
                n= input.nextInt(); check= true;
            }catch(Exception e){
                System.out.println("phai nhap so, moi nhap lai"); input.nextLine();
            }
        }
        return (n);
    }
    public static int New(int n){ int T=0;
        while(n>0){
            T+= n%10;
            n/= 10;
        }
        return (T);
    }
    //Ham kiem tra so nguyen to
    public static boolean incheck(int n){ if(n>1){for(int i=2;i<=Math.sqrt(n);i++){ if(n%i==0) return false;
    }
        return true;
    }
    else return false;
    }
    public static void result(int n){
        int i=2; while(n>1){
            if(incheck(i)){
                if(n%i==0){
                    System.out.print(i+"."); n/=i;
                }
                else i++;
            }
            else i++;
        }
    }
    public static void main(String[] args) {
        System.out.print("Nhap n");
        int n= process(); System.out.print("n= " ); result(n);
        System.out.println("Tong cac chu so cua "+n+" la: "+New(n));
    }
}

File: app/src/at-tamle/java/asiantech/internship/summer/kotlin/TotalAnalytical.kt
package asiantech.internship.summer.kotlin

import java.util.Scanner
import kotlin.math.sqrt

object TotalAnalytical {
    private fun process(): Int {
        val input = Scanner(System.`in`)
        var check = false
        var n = 0
        while (!check) {
            try {
                n = input.nextInt()
                check = true
            } catch (e: Exception) {
                println("Ban phai nhap so! hay nhap lai...")
                input.nextLine()
            }

        }
        return n
    }

    private fun input(n: Int): Int {
        var n = n
        var k = 0
        while (n > 0) {
            k += n % 10
            n /= 10
        }
        return k
    }

    private fun inCheck(n: Int): Boolean {
        if (n > 1) {
            var i = 2
            while (i <= sqrt(n.toDouble())) {
                if (n % i == 0) return false
                i++
            }
            return true
        } else
            return false
    }

    private fun result(n: Int) {
        var n = n
        var i = 2
        while (n > 1) {
            if (inCheck(i)) {
                if (n % i == 0) {
                    print("$i.")
                    n /= i
                } else
                    i++
            } else
                i++
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        print("Nhap n")
        val n = process()
        print("n= ")
        result(n)
        println("Tong cac chu so cua $n la: " + input(n))
    }
}
